echopop/workflow/feat_hake.py

```diff
@@ -163,124 +168,482 @@
+# ---- can be removed if it ends up never being used
+age_distribution: pd.DataFrame = utils.binned_distribution(age_bins)
+length_distribution: pd.DataFrame = utils.binned_distribution(length_bins)
 
-root_path = "WHERE_ALL_DATA_ARE"
-species_code = "SPECIES_CODE"
-df_nasc_no_age1: pd.DataFrame  # extracted nasc data from above, can also be df_nasc_all_ages
+# ==================================================================================================
+# 'Binify' the biological data [age, length]
+# ------------------------------------------
+# Note: this argument can be done 'inplace'
+data: Union[pd.DataFrame, Dict[str, pd.DataFrame]] = dict_df_bio_ks 
+bin_distribution: pd.DataFrame = length_distribution.copy()
+bin_column: str = "length"
+inplace: bool = False
 
-bio_path_dict: dict  # the "biological" section of year_config.yml
-# this will be simplified now that we read from the master spreadsheet
-strata_path_dict: dict  # the "stratification" section of year_config.yml
+# Outputs:
+# ---- `Union[pd.DataFrame, Dict[str, pd.DataFrame], None]`
+# ---- `pandas.DataFrame` if a single `pandas.DataFrame` input
+# ---- The same structured `Dict[str, pandas.DataFrame]` if a `Dict[str, pandas.DataFrame]` input
+# ---- `None` when `inplace=True`
 
-dict_df_bio = load_data.load_biological_data(root_path, bio_path_dict, species_code)
-dict_df_strata = load_data.load_stratification(root_path, strata_path_dict)
+# Length
+dict_df_bio_binned_ks: Dict[str, pd.DataFrame] = utils.binify(
+    data, bin_distribution, bin_column, inplace
+)
 
-# Consolidate all input data into df_acoustic_dict
-df_nasc_no_age1 = load_data.consolidate_all_data(
-    df_nasc=df_nasc_no_age1, df_bio_dict=dict_df_bio, df_strata_dict=dict_df_strata
+# Age
+utils.binify(
+    data=dict_df_bio_binned_ks, bin_distribution=age_distribution, bin_column="age", inplace=True
 )
 
+# ==================================================================================================
+# Fit length-weight regression to the binned data
+# -----------------------------------------------
+length_weight_dataset: pd.DataFrame = dict_df_bio_binned_ks["specimen"].copy()
 
-# ===========================================
-# Compute biological composition based on stratum
-length_bins: np.array  # length bin specification
-df_length_weight, df_regression = get_proportions.length_weight_regression(
-    dict_df_bio["specimen"], length_bins
+# Outputs: 
+# ---- `pandas.Series` that includes the fitted slope and intercepts for the length-weight 
+# ---- log-linear regressions. This can either be ran directly on a `pandas.DataFrame` or via 
+# ---- `pandas.DataFrame.groupby`
+length_weight_coefs_all: pd.Series = biology.fit_length_weight_regression(length_weight_dataset)
+length_weight_coefs_sex = length_weight_dataset.groupby(["sex"]).apply(
+    biology.fit_length_weight_regression,
+    include_groups=False
```

Comment:
You don't need to make a new data variable `length_weight_dataset`, just pass `dict_df_bio_binned_ks["specimen"]` directionly in to the `fit_length_weight_regression` function.

Use a dictionary `length_weight_coefs` with keys "all" and "sex" to store the length-weight relationship. It helps to make the flow clean by consolidating things like this.